src/shogun/io/ARFFFile.h

```diff
@@ -0,0 +1,545 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_ARFFFILE_H
+#define SHOGUN_ARFFFILE_H
+
+#include <shogun/base/init.h>
+#include <shogun/base/variant.h>
+#include <shogun/features/CombinedFeatures.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/lib/SGVector.h>
+
+#include <cctype>
+#include <fstream>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace shogun
+{
+	/** Contains miscellaneous string manipulation functions using the STL and
+	 * Java to C++ date format utilities */
+	namespace arff_detail
+	{
+		/**
+		 * Checks if string is blank
+		 * @param line to check
+		 * @return bool whether line is empty
+		 */
+		SG_FORCED_INLINE bool is_blank(const std::string& line)
```

Comment:
stupid question, isnt there some minimal lib to do these things?
Since technically all this should be tested, or?

Reply:
I am not sure if there is such a library.. But these are quite generic functions that should probably be moved to another file

Reply:
++

Reply:
I actually wonder whether it even might make sense to SWIG a few of those ... but I am not 100% sure. What do you think?

Reply:
probably there will be a lot of name clashes ...

Reply:
but moving to a sep file is a good idea to keep this one shorter

Reply:
I don't think they should be in swig, they are so short that it is better for users to write their own implementations of these functions in their own language no?

Reply:
yeah ok